File: android/app/src/main/java/com/sumitrack/android/domain/models/Payment.kt
package com.sumitrack.android.domain.models

import java.math.BigDecimal
import java.time.Instant

data class Payment(
    val id: String,
    val fkTenant: String,
    val fkSale: String,
    val fkInstallment: String?,
    val method: PaymentMethodType,
    val amount: BigDecimal,
    val paidAt: Instant,
    val createdAt: Instant,
    val updatedAt: Instant,
    val syncStatus: SyncStatus,
)

File: backend/src/Sumitrack.Api/Models/Entities/Payment.cs
namespace Sumitrack.Api.Models.Entities;

public class Payment
{
    public Guid Id { get; set; }
    public Guid FkTenant { get; set; }
    public Guid FkSale { get; set; }
    public Guid? FkInstallment { get; set; }
    public string Method { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public DateTime PaidAt { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public string SyncStatus { get; set; } = "synced";
}
